src/Algebra/Bundles/Pointed.agda

```diff
@@ -0,0 +1,50 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- `Pointed` intermediate between `Monoid` and `SemiringWithoutAnnihilatingZero`
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Algebra.Bundles
+open import Algebra.Bundles.Raw
+open import Algebra.Core
+open import Algebra.Structures.Pointed as Pointed using (IsPointedMonoid)
+import Algebra.Properties.Monoid.Mult as Mult
+open import Data.Nat.Base as ℕ using (ℕ)
+open import Data.Unit.Base
+open import Level using (Level; suc; _⊔_; Lift)
+open import Relation.Binary.Core using (Rel)
+
+module Algebra.Bundles.Pointed where
+
+private
+  variable
+    c ℓ : Level
+
+------------------------------------------------------------------------
+-- Bundles with 1 binary operation & 2 elements
+------------------------------------------------------------------------
+
+record PointedMonoid c ℓ : Set (suc (c ⊔ ℓ)) where
+  field
+    rawMonoid  : RawMonoid c ℓ
+  open RawMonoid rawMonoid using (Carrier)
+  field
+    •      : Carrier
```

Comment:
The comment "1 binary operation & 2 elements" is correct, but the rest of the names (`Pointed`, etc) give the wrong impression of having a single distinguished element. `BiPointed` ?

Reply:
Much to think about here. Ever since my abortive `Pointed` draft PR, I've been thinking about `PointedX' (and how to formalise the map from `X` in a 'signature polymorphic' way) in general...

The two points in `PointedMonoid` play distinct roles, so I'm unhappy to identify them as '2 points', because one is distinguished as an identity, while the other is an unconstrained extension of the signature... 

... `BiPointedSet`/`BiPointedMagma` do make sense, especially as `Raw` bundles, but I've always been resistant to universal algebra ideas in this regard precisely because not all 'freely constructed' such signatures make sense in the absence of additional structure...? 

So from my POV, I'd want to keep things more cleanly separated...

Reply:
Nitpick: they do "make sense", but they are not necessarily *sufficiently useful*. Phrased that way, I'd say that our current state of knowledge makes this true. [Usefulness is a temporal quality, not absolute.]

Reply:
The next question is: that extra point, is it meant to be preserved by homomorphisms or not? If not, then rather than `Pointed` what you're describing is `Inhabited` (I'm writing that up separately, i.e. the observation that the category whose objects are pairs of a type and a point of that type, but whose homomorphisms don't preserve that point, are exactly the inhabited types, without needing to have a notion of truncation).

Of course, a `Monoid` is always inhabited -- the right name here would be `NonTrivialMonoid` although that's tricky as the one-point monoid is a model as you don't actually want to assert that the extra point is not unit!  Best you might be able to do is to ask that the obvious homomorphism induced by 'swap' not be homotopic to the identity.

Reply:
Also an excellent observation (although I'm still insufficently homotopically-inclined to arbitrate on the last point). I think that points should be preserved (esp. in this setting, because `_+ pt` is the unary operation at hand, and we want it as (part of) an NNO-instance to be preserved?

Reply:
And `NonTrivialMonoid`...: does that exclude the case `0# = 1#`? I wouldn't want to insist on that...?